src/ecl/saveData.ts

```diff
@@ -0,0 +1,250 @@
+            await this.fetchCSVRows(result, start, number);
+        else
+            await this.fetchExcelRows(this._sheet, result, start, number);
+    }
+
+    async fetchCSVRows(result, start: number, number: number) {
+        await result.fetchRows(start, number).then(rows => {
+            if (start == 0) {
+                const header = [];
+                for (const row of rows) {
+                    for (const key in row) {
+                        header.push(key);
+                    }
+                    break;
+                }
+                const filePath = this._saveType == SaveType.RESULT ? this._filePath : this.filePathResult(this._filePath, result.Name, ".csv");
+                    
+                this._csvWriter = this._createCsvWriter({
+                    path: filePath,
+                    header
+                });
+            }
+
+            for (const row of rows) {
+                const rowStr = [];
+                for (const key in row) {
+                    rowStr.push(row[key]);
+                }
+                this._dataRows.push(rowStr);
+            }
+        });
+    }
+
+    async fetchExcelRows(sheet, result, start: number, number: number) {
+        await result.fetchRows(start, number).then(rows => {
+            let firstrow = 1;
+            const columns = [];
+            rows.forEach(function (row) {
+                if (firstrow) {
+                    for (const key in row) {
+                        const col = {};
+                        col["header"] = key;
+                        col["key"] = key;
+                        columns.push(col);
+                    }
+                    sheet.columns = columns;
+                    firstrow = 0;
+                }
+                sheet.addRow(row);
+            });
+        });
+    }
+    
+    async askRowCount(results): Promise<number> {
+        let maxRowsFound = 0;
+
+        for (const result of results) {
+            if (result.Total > maxRowsFound)
+            maxRowsFound = result.Total;
+        }
+        let userLimit = -1;
+        if (maxRowsFound > this._maxRows) {
+            userLimit = await this.promptUserMaxRows(maxRowsFound);
+        } else {
+            userLimit = this._maxRows;
+        }
+        return userLimit;
+    }
+
+    async promptUserMaxRows(rowCount: number): Promise<number> {
+        let userLimit = -1;
+        while (userLimit == -1 || userLimit > this._maxRows) {
+            userLimit = await this.promptMaxRows(rowCount);
+        }
+        return userLimit;
+    }
+
+    async promptMaxRows(maxRowsFound: number): Promise<number> {
+        if (maxRowsFound <= this._maxRows)
+            return maxRowsFound;
+        const rows = await vscode.window.showInputBox({
+            title: localize("Max number of rows to save"),
+            prompt: `${maxRowsFound} ${localize("rows found. Please choose")} ${this._maxRows} ${localize("or less")}`,
```

Comment:
For future ref:  Another option is to have a single string "MMM rows found. Please choose NNN or less" and swap out the MMM and NNN with numbers (can be easier for the translators).